frontend/src/components/CreateInstanceForm.vue

```diff
@@ -410,6 +432,9 @@ const changeInstanceEngine = (engine: EngineType) => {
       state.instance.host = isDev() ? "127.0.0.1" : "host.docker.internal";
     }
   }
+  if (engine !== "POSTGRES") {
+    state.instance.database = "";
```

Comment:
I would choose to set instance.database = "" on sending the request (either at handleWarningModalOkClick or doCreate), so switching engine back and forth will not reset the field.

Reply:
Updated. PTAL